<?xml version="1.0"?>

<launch>


  <!-- FCU communication -->
  <node name="rosflight_io" pkg="rosflight" type="rosflight_io" output="screen">
    <param name="udp" value="false"/>
    <param name="port" value="/dev/ttyACM0"/>
  </node>
  
  <!-- GPS Node-->
  <node name="f9p" pkg="ublox" type="ublox_ros">
    <param name="serial_port" value="/dev/ttyACM1"/>
    <!-- setting up the module as a base to recieve GPS (3 is rover)-->
    <param name="rtk_type" value="2"/>
  </node> 
  
  <!-- Load common parameters -->
  <rosparam command="load" file="$(find roscopter)/params/F450.yaml"/>

  <!-- PID Position Controller -->
  <node name="controller" pkg="roscopter" type="controller">
    <remap from="estimate" to="odom"/>
  </node>

  <!-- Estimator -->
  <rosparam command="load" file="$(find roscopter)/params/ekf.yaml" ns="estimator"/>
  <node name="estimator" type="ekf_node" pkg="roscopter" output="screen">
    <remap from="imu" to="/imu/data"/>
    <remap from="baro" to="/baro"/>
    <remap from="ublox_gnss" to="/PosVelEcef"/>
    <remap from="pose" to="       "/>

  </node>

  <!-- Waypoint Manager -->
  <node name="waypoint_manager" pkg="roscopter" type="waypoint_manager.py" output="screen">
    <remap from="waypoint" to="high_level_command"/>
    <remap from="state" to="odom"/>
  </node>

</launch>
